amps/ags/rm-community/rm-community-repo/unit-test/java/org/alfresco/module/org_alfresco_module_rm/disposition/RetentionScheduleModelUnitTest.java

```diff
@@ -0,0 +1,121 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.module.org_alfresco_module_rm.disposition;
+
+import org.alfresco.module.org_alfresco_module_rm.test.util.BaseUnitTest;
+import org.alfresco.rm.rest.api.impl.ApiNodesModelFactory;
+import org.alfresco.rm.rest.api.model.RetentionSchedule;
+import org.alfresco.rm.rest.api.model.RetentionScheduleActionDefinition;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.Period;
+import org.alfresco.service.namespace.QName;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Api nodes modal factory unit test
+ */
+public class RetentionScheduleModelUnitTest extends BaseUnitTest
+{
+    private static final String AUTHORITY = "authority";
+    private static final String INSTRUCTIONS = "instructions";
+    private static final String RETAIN_STEP = "retain";
+
+    @InjectMocks
+    private ApiNodesModelFactory apiNodesModelFactory;
+
+    @Test
+    public void mapRetentionScheduleDataTest()
+    {
+        // Mock data
+        DispositionSchedule dispositionSchedule = mock(DispositionSchedule.class);
+        NodeRef nodeRef = mock(NodeRef.class);
+        when(dispositionSchedule.getDispositionAuthority()).thenReturn(AUTHORITY);
+        when(dispositionSchedule.getDispositionInstructions()).thenReturn(INSTRUCTIONS);
+        when(dispositionSchedule.getNodeRef()).thenReturn(nodeRef);
+        when(dispositionSchedule.isRecordLevelDisposition()).thenReturn(false);
+        when(apiNodesModelFactory.mapRetentionScheduleData(dispositionSchedule)).thenReturn(mock(RetentionSchedule.class));
+
+        // Call the method
+        RetentionSchedule result = apiNodesModelFactory.mapRetentionScheduleData(dispositionSchedule);
+
+        assertEquals(dispositionSchedule.getNodeRef().getId(),result.getId());
+        assertEquals(dispositionSchedule.getDispositionAuthority(),result.getAuthority());
+        assertEquals(dispositionSchedule.getDispositionInstructions(),result.getInstructions());
+        assertEquals(dispositionSchedule.isRecordLevelDisposition(),result.isRecordLevel());
+    }
+
+    @Test
+    public void mapRetentionScheduleActionDefDataTest()
+    {
+        // Mock data
+        DispositionActionDefinition dispositionActionDefinition = mock(DispositionActionDefinition.class);
+        NodeRef nodeRef = mock(NodeRef.class);
+        Period mockPeriod = mock(Period.class);
+        QName mockQName = mock(QName.createQName("qname"));
```

Comment:
Is there a benefit here over using a real QName?
```suggestion
        QName mockQName = QName.createQName("qname");
```